apps/src/javalab/javalabRedux.js

```diff
@@ -36,17 +37,31 @@ export const renameFile = (oldFilename, newFilename) => ({
   newFilename
 });
 
-export const setSource = (filename, source, isVisible = true) => ({
+export const setSource = (
+  filename,
+  source,
+  isVisible = true,
+  isValidation = false
+) => ({
   type: SET_SOURCE,
   filename,
   source,
-  isVisible
+  isVisible,
+  isValidation
 });
 
-export const sourceVisibilityUpdated = (filename, isVisible) => ({
+export const sourceVisibilityUpdated = (filename, isVisible, isValidation) => ({
```

Comment:
thoughts on this approach:  it may be confusing in the future as to why we have two functions that have different names but do the same thing. I think it may be cleaner to keep the two separate functions: `sourceVisibilityUpdated`, `sourceValidationUpdated`, but only take the relevant boolean (`isVisible` or `isValdation`). Downside is you have to call both functions when making an update, but it makes it clearer what each function is doing.

Reply:
Okay sounds good- I'll switch back over to this! Thank you!

Reply:
Updated! Let me know if this looks right to you!